src/cpp/rediscluster.cpp

```diff
@@ -126,6 +126,43 @@ CommandReply RedisCluster::run(AddressAnyCommand &cmd)
     return _run(cmd, _last_prefix);
 }
 
+// Run a non-keyed Command that addresses any db node on the server
+CommandReply RedisCluster::run(AddressAllCommand &cmd)
+{
+    // Bounds check the key_index
+    if (cmd.key_index != -1 && cmd.get_field_count() < cmd.key_index) {
+        throw SRInternalException("Invalid key_index executing command!");
+    }
+
+    // Find the segment to be tweaked for each node
+    std::string field;
+    if (cmd.key_index != -1) {
+        Command::const_iterator it = cmd.cbegin();
+        int i = 0;
+        for ( ; it != cmd.cend(); it++) {
```

Comment:
Coming from a place of relatively little knowledge, do we only expect one field to be set in a given command? Otherwise this loop as written seems like it would just return the last one even if it detects more than one.

Reply:
Yes, there should only be a single key